oap-server/server-library/library-server/src/main/java/org/apache/skywalking/oap/server/library/server/grpc/GRPCServer.java

```diff
@@ -90,12 +93,26 @@ public String serverClassify() {
     @Override
     public void initialize() {
         InetSocketAddress address = new InetSocketAddress(host, port);
-
+        ArrayBlockingQueue blockingQueue = new ArrayBlockingQueue(10000);
+        ExecutorService executor = new ThreadPoolExecutor(500, 500, 60,
+                TimeUnit.SECONDS, blockingQueue, new CustomThreadFactory("grpcServerPool"), new CustomRejectedExecutionHandler());
         nettyServerBuilder = NettyServerBuilder.forAddress(address);
-        nettyServerBuilder = nettyServerBuilder.maxConcurrentCallsPerConnection(maxConcurrentCallsPerConnection).maxMessageSize(maxMessageSize);
+        nettyServerBuilder = nettyServerBuilder.maxConcurrentCallsPerConnection(maxConcurrentCallsPerConnection).maxMessageSize(maxMessageSize).executor(executor);
         logger.info("Server started, host {} listening on {}", host, port);
     }
 
+    static class CustomRejectedExecutionHandler implements RejectedExecutionHandler {
+        private final AtomicInteger rejectNumber = new AtomicInteger(1);
+
+        @Override
+        public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
+            if (rejectNumber.getAndIncrement() == 100) {
+                logger.warn("Grpc server thread pool is full, rejecting the task");
+                rejectNumber.set(1);
```

Comment:
If the rejectNumber is just for logging, no need to keep it so accurate. A simple int w/ volatile is enough. `AtomicInteger` causes unnecessary CPU.

Reply:
good idea